src/ssh/sftp_client.cpp

```diff
@@ -16,189 +16,309 @@
  */
 
 #include <multipass/ssh/sftp_client.h>
-#include <multipass/ssh/throw_on_error.h>
-#include <multipass/utils.h>
 
 #include "ssh_client_key_provider.h"
-
+#include <multipass/file_ops.h>
 #include <multipass/format.h>
+#include <multipass/ssh/sftp_utils.h>
+#include <multipass/ssh/throw_on_error.h>
+#include <multipass/utils.h>
 
 #include <array>
-#include <cassert>
 #include <fcntl.h>
 
-#include <QFile>
-
-namespace mp = multipass;
-
-namespace
-{
-// TODO: For push/pull, use actual file permissions
 constexpr int file_mode = 0664;
 constexpr auto max_transfer = 65536u;
-const std::string stream_file_name{"stream_output.dat"};
+const char* stream_file_name = "stream_output.dat";
```

Comment:
Just curious why the type had to be changed here.

Reply:
honestly, it didn't have to be changed, it just bothered me that the string was just long enough to not qualify for a small string optimization, and there was no point in having it allocated on the heap, but in the end i guess the compiler probably optimized the whole thing anyway. i'll change it back :)